lte/gateway/python/magma/mobilityd/ip_allocator_dhcp.py

```diff
@@ -24,9 +24,10 @@
 from copy import deepcopy
 from ipaddress import ip_address, ip_network
 from threading import Condition
-from typing import List
+from typing import List, Optional, cast
 
 from magma.mobilityd.ip_descriptor import IPDesc, IPState, IPType
+from typing_extensions import TypeGuard
```

Comment:
I didn't know mypy had introduced type guards. :partying_face: 

We'll probably need the typing_extensions package in `setup.py` to use it though.

Reply:
Looks like I was wrong, no additional dependency needed.

Reply:
@sebathomas seems like we do need it, on .deb magma package there's an issue due to the missing dependency